extensions/amp-audio/1.0/test/test-amp-audio.js

```diff
@@ -0,0 +1,385 @@
+      expect(audio.childNodes[0].tagName).to.equal('SOURCE');
+      expect(audio.childNodes[0].getAttribute('src')).to.equal('audio.mp3');
+      expect(audio.childNodes[1].tagName).to.equal('SOURCE');
+      expect(audio.childNodes[1].getAttribute('src')).to.equal('audio.ogg');
+      expect(audio.childNodes[2].nodeType).to.equal(Node.TEXT_NODE);
+      expect(audio.childNodes[2].textContent).to.equal('Unsupported.');
+    });
+
+    it('should set its dimensions to the browser natural', async () => {
+      element = attachAndRun({
+        src: 'audio.mp3',
+      });
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Fetch audio element from shadowRoot
+      const {shadowRoot} = element;
+      const audio = shadowRoot.querySelector('audio');
+
+      expect(element.style.width).to.be.equal('300px');
+      expect(element.style.height).to.be.equal('30px');
+      if (/Safari|Firefox/.test(navigator.userAgent)) {
+        // Safari has default sizes for audio tags that cannot
+        // be overridden.
+        return;
+      }
+      expect(audio.offsetWidth).to.be.equal(300);
+      expect(audio.offsetHeight).to.be.equal(30);
+    });
+
+    it('should set its natural dimension only if not specified', async () => {
+      element = attachAndRun({
+        'width': '500',
+        src: 'audio.mp3',
+      });
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      expect(element.style.width).to.be.equal('500px');
+      expect(element.style.height).to.be.equal('30px');
+    });
+
+    it('should fallback when not available', async () => {
+      // For this single test, cause audio elements that are
+      // created to lack the necessary feature set, which should trigger
+      // fallback behavior.
+      const {createElement} = doc;
+      doc.createElement = (name) => {
+        if (name === 'audio') {
+          name = 'busted-audio';
+        }
+        return createElement.call(doc, name);
+      };
+
+      const element = doc.createElement('div');
+      element.toggleFallback = env.sandbox.spy();
+      const audio = new AmpAudio(element);
+      // audio.buildAudioElement();
+      await audio.layoutCallback();
+      expect(element.toggleFallback).to.be.calledOnce;
+    });
+
+    it('should propagate ARIA attributes', async () => {
+      element = attachAndRun({
+        src: 'audio.mp3',
+        'aria-label': 'Hello',
+        'aria-labelledby': 'id2',
+        'aria-describedby': 'id3',
+      });
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Fetch audio element from shadowRoot
+      const {shadowRoot} = element;
+      const audio = shadowRoot.querySelector('audio');
+
+      expect(audio.getAttribute('aria-label')).to.equal('Hello');
+      expect(audio.getAttribute('aria-labelledby')).to.equal('id2');
+      expect(audio.getAttribute('aria-describedby')).to.equal('id3');
```

Comment:
This only works if you propagate in `base-element.js` and `component.js`:
```js
// base-element.js
BaseElement['props'] = {
  'aria-label': {
    attr: 'aria-label',
    type: 'string',
  },
  'aria-labelledby': {
    attr: 'aria-labelledby',
    type: 'string',
  },
  'aria-describedby': {
    attr: 'aria-describedby',
    type: 'string',
  },
  ...
}
```

```js
// component.js
const {
  ...
  'aria-describedby': ariaDescribedby,
  'aria-label': ariaLabel,
  'aria-labelledby': ariaLabelledby,
  ...
} = props;

...

return <audio ...
      aria-label={ariaLabel}
      aria-labelledby={ariaLabelledby}
      aria-describedby={ariaDescribedby}>
```

Reply:
Thank you so much for this correction 🙂

Reply:
Above code works well 🙌🏻

Just curious, the following did not work:
```javascript
// base-element.js
BaseElement['props'] = {
  ariaLabel: {
    attr: 'aria-label',
    type: 'string',
  },
  ariaLabelledBy: {
    attr: 'aria-labelledby',
    type: 'string',
  },
  ariaDescribedBy: {
    attr: 'aria-describedby',
    type: 'string',
  },
  ...
}
```

Whenever I use `'aria-describedby'`, it works but not when used directly as `ariaDescribedby`.
Note: If `'aria-describedby'` is mentioned in `base-element.js`, `'aria-describedby': ariaDescribedby` is mentioned in `component.js`. And if  `ariaDescribedby`  is mentioned in `base-element.js` then `ariaDescribedby` is mentioned in `component.js`.